code/game/machinery/doors/airlock.dm

```diff
@@ -48,253 +55,23 @@
-	var/last_event = 0
-
-/obj/machinery/door/airlock/uranium/process()
-	if(world.time > last_event+20)
-		if(prob(50))
-			radiate()
-		last_event = world.time
+/obj/machinery/door/airlock/New()
 	..()
-
-/obj/machinery/door/airlock/uranium/proc/radiate()
-	for(var/mob/living/L in range (3,src))
-		L.irradiate(15)
-	return
-
-/obj/machinery/door/airlock/plasma
-	name = "plasma airlock"
-	desc = "No way this can end badly."
-	icon = 'icons/obj/doors/Doorplasma.dmi'
-	var/mineral = "plasma"
-	doortype = /obj/structure/door_assembly/door_assembly_plasma
-
-/obj/machinery/door/airlock/plasma/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
-	if(exposed_temperature > 300)
-		PlasmaBurn(exposed_temperature)
-
-/obj/machinery/door/airlock/plasma/proc/ignite(exposed_temperature)
-	if(exposed_temperature > 300)
-		PlasmaBurn(exposed_temperature)
-
-/obj/machinery/door/airlock/plasma/proc/PlasmaBurn(temperature)
-	atmos_spawn_air(SPAWN_HEAT | SPAWN_TOXINS, 500)
-	new/obj/structure/door_assembly/door_assembly_0( src.loc )
-	qdel(src)
-
-/obj/machinery/door/airlock/plasma/BlockSuperconductivity() //we don't stop the heat~
-	return 0
-
-/obj/machinery/door/airlock/clown
-	name = "bananium airlock"
-	desc = "Honkhonkhonk"
-	icon = 'icons/obj/doors/Doorbananium.dmi'
-	var/mineral = "bananium"
-	doortype = /obj/structure/door_assembly/door_assembly_clown
-
-/obj/machinery/door/airlock/sandstone
-	name = "sandstone airlock"
-	icon = 'icons/obj/doors/Doorsand.dmi'
-	var/mineral = "sandstone"
-	doortype = /obj/structure/door_assembly/door_assembly_sandstone
-
-/obj/machinery/door/airlock/science
-	icon = 'icons/obj/doors/Doorsci.dmi'
-	doortype = /obj/structure/door_assembly/door_assembly_science
-
-/obj/machinery/door/airlock/glass_science
-	name = "glass airlock"
-	icon = 'icons/obj/doors/Doorsciglass.dmi'
-	opacity = 0
-	doortype = /obj/structure/door_assembly/door_assembly_science/glass
-	glass = 1
-
-/obj/machinery/door/airlock/highsecurity
-	name = "high tech security airlock"
-	icon = 'icons/obj/doors/hightechsecurity.dmi'
-	doortype = /obj/structure/door_assembly/door_assembly_highsecurity
-	explosion_block = 2
-
-/obj/machinery/door/airlock/shuttle
-	name = "shuttle airlock"
-	icon = 'icons/obj/doors/doorshuttle.dmi'
-	doortype = /obj/structure/door_assembly/door_assembly_shuttle
-
-/obj/machinery/door/airlock/wood
-	name = "wooden airlock"
-	icon = 'icons/obj/doors/Doorwood.dmi'
-	var/mineral = "wood"
-	doortype = /obj/structure/door_assembly/door_assembly_wood
-
-/obj/machinery/door/airlock/virology
-	icon = 'icons/obj/doors/Doorviro.dmi'
-	doortype = /obj/structure/door_assembly/door_assembly_viro
-
-/obj/machinery/door/airlock/glass_virology
-	icon = 'icons/obj/doors/Doorviroglass.dmi'
-	opacity = 0
-	doortype = /obj/structure/door_assembly/door_assembly_viro/glass
-	glass = 1
+	wires = new(src)
+	if(src.closeOtherId != null)
+		spawn (5)
+			for (var/obj/machinery/door/airlock/A in world)

```

Comment:
There is no could about it, you touched it, it's your duty to fix the ```in world``` now.
There's a machine list for definite, but there might also be an airlock one, not sure.